de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/server/rpl/exp/RabbitMqExportProcessor.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2015 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+package org.adempiere.server.rpl.exp;
+
+import de.metas.logging.LogManager;
+import org.adempiere.process.rpl.IExportProcessor;
+import org.apache.commons.lang3.StringUtils;
+import org.compiere.model.I_EXP_ProcessorParameter;
+import org.compiere.model.MEXPProcessor;
+import org.compiere.model.X_EXP_ProcessorParameter;
+import org.compiere.util.Trx;
+import org.slf4j.Logger;
+import org.springframework.amqp.core.BindingBuilder;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.core.TopicExchange;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.w3c.dom.Document;
+
+import javax.xml.transform.*;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.Properties;
+
+/**
+ * Send AMQP Messages
+ */
+public class RabbitMqExportProcessor implements IExportProcessor
+{
+
+	/**
+	 * Name of the exchange name parameter
+	 */
+	private static final String EXCHANGE_NAME_PARAMETER = "exchangeName";
+	/**
+	 * Name of the routing key parameter
+	 */
+	private static final String ROUTING_KEY_PARAMETER = "routingKey";
+	/**
+	 * Name of the is durable queue parameter
+	 */
+	private static final String IS_DURABLE_QUEUE_PARAMETER = "isDurableQueue";
+	/**
+	 * Logger
+	 */
+	protected Logger log = LogManager.getLogger(getClass());
+
+	@Override
+	public void process(Properties ctx, MEXPProcessor expProcessor, Document document, Trx trx)
+			throws Exception
+	{
+		String host = expProcessor.getHost();
+		int port = expProcessor.getPort();
+		String account = expProcessor.getAccount();
+		String password = expProcessor.getPasswordInfo();
+		String exchangeName = StringUtils.EMPTY;
+		String routingKey = StringUtils.EMPTY;
+		boolean isDurableQueue = true;
+
+		// Read all processor parameters and set them!
+		I_EXP_ProcessorParameter[] processorParameters = expProcessor.getEXP_ProcessorParameters();
+		if (processorParameters != null && processorParameters.length > 0)
```

Comment:
avoid unneeded nesting; see if we can make sure that `expProcessor.getEXP_ProcessorParameters()` never returns `null` so we can remove this `if`